### PR TITLE
Add product and recent_view_count to Dashboard APIs

### PR DESCRIPTION
**Pull request by jinhyukchang**
### Summary of Changes

Updated:
 - Added `product` and `recent_view_count` to Dashboard Detail GET API and Bookmarks GET API
 - Neo4j Cypher query to perform early aggregation for better performance.

### Tests

Updated unit tests. Performed integration tests.

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

File: metadata_service/api/swagger_doc/template.yml [hunk 1]
```diff
@@ -308,6 +312,10 @@ components:
             description: 'Badges associated with the Dashboard'
             items:
               $ref: '#/components/schemas/TagFields'
+        recent_view_count:
+          type: int
+          description: 'Dashboard recent view count'
```
Comment (feng-tao) [inline metadata_service/api/swagger_doc/template.yml:317]: the recent_view_count name is a bit confusing? is it for last two weeks?

Comment (jinhyukchang) [reply metadata_service/api/swagger_doc/template.yml:317]: > the recent_view_count name is a bit confusing? is it for last two weeks?

Yeah, it is actually intentional as it's hard to make this value precise for it is relative value. In other words, actual `recent` value (either 2 weeks or 4 weeks) depends on how accurately `databuilder` ingests and remove stale data. To clarify why I used `recent` here is to show this is not a total view count. WDYT?

Comment (feng-tao) [reply metadata_service/api/swagger_doc/template.yml:317]: yeah, I understand that part, try to see if we could have more precise term. Maybe you could add a comment to make that explict that it depends on databuilder ingestion?

Comment (jinhyukchang) [reply metadata_service/api/swagger_doc/template.yml:317]: Sure, will do!

**Pull request merged**